code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/listener/SystemStackUpdate.java

```diff
@@ -137,13 +145,44 @@ protected boolean addStack(Account account, String stack) throws IOException {
     }
 
     protected String getFile(String stack, String filename) throws IOException {
-        String resource = String.format(STACK_RESOURCE, stack, filename);
-        InputStream is = getClass().getResourceAsStream(resource);
-        if (is == null) {
+ 
+        if(stack != null && stack.equalsIgnoreCase("publicdns")) {
+            String catalogTemplateUrl = String.format(CATALOG_RESOURCE_URL, stack);
+            //get the latest version from the catalog template
+            Request temp = Request.Get(catalogTemplateUrl);
+            Response res = temp.execute();
+
+            Template template = jsonMapper.readValue(res.returnContent().asBytes(), Template.class);
+
+            if (template.getDefaultVersion() != null && template.getVersionLinks() != null) {
+
+                String versionUrl = template.getVersionLinks().get(template.getDefaultVersion());
+                log.debug("Catalog system template versionUrl is: [{}]", versionUrl);
+                Request versionReq = Request.Get(versionUrl);
+                Response versionRes = versionReq.execute();
+
+                Template templateVersion = jsonMapper.readValue(versionRes.returnContent().asBytes(), Template.class);
+
+                if(templateVersion != null) {
+                    if(filename!= null && filename.startsWith("docker-")){
+                        return templateVersion.getDockerCompose();
+                    }else if(filename!= null && filename.startsWith("rancher-")){
+                        return templateVersion.getRancherCompose();
+                    }
+                }
+                log.error("Failed to load template version");
+            } else {
+                log.error("Failed to load template details");
+            }
             return null;
+        } else {
+            String resource = String.format(STACK_RESOURCE, stack, filename);
```

Comment:
We don't need fallback logic.  We will just always go to the catalog now.  The existing swarm, k8s functionality is not currently enabled so you don't need to worry about breaking things.